pkg/service/model.go

```diff
@@ -13,15 +16,81 @@ import (
 	"github.com/instill-ai/artifact-backend/pkg/constant"
 	"github.com/instill-ai/artifact-backend/pkg/logger"
 
+	"encoding/base64"
+	"sync"
+
 	artifactpb "github.com/instill-ai/protogen-go/artifact/artifact/v1alpha"
 	modelpb "github.com/instill-ai/protogen-go/model/model/v1alpha"
+
+	"github.com/pdfcpu/pdfcpu/pkg/api"
 )
 
 const ConvertDocToMDModelID = "docling"
 const ConvertDocToMDModelVersion = "v0.1.0"
 
+// splitPDFIntoBatches splits a PDF into batches of n pages each and returns a slice of base64-encoded sub-PDFs
+func splitPDFIntoBatches(pdfData []byte, batchSize int) ([]string, error) {
+	// Create a temporary directory
+	tempDir, err := os.MkdirTemp("", "pdfcpu-*")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp directory: %w", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Prevent pdfcpu from trying to write out $HOME/.config/pdfcpu
+	api.DisableConfigDir()
+
+	// Write the input PDF to a temporary file
+	inputFile, err := os.CreateTemp(tempDir, "input-*.pdf")
+	if err != nil {
+		return nil, err
+	}
+	inputFilePath := inputFile.Name()
+	if _, err := inputFile.Write(pdfData); err != nil {
+		inputFile.Close()
+		return nil, err
+	}
+	inputFile.Close()
+	defer os.Remove(inputFilePath)
+
+	// Read number of pages
+	ctx, err := api.ReadContextFile(inputFilePath)
+	if err != nil {
+		return nil, err
+	}
+	nPages := ctx.PageCount
+
+	var batches []string
+	for i := 0; i < nPages; i += batchSize {
+		end := i + batchSize
+		if end > nPages {
+			end = nPages
+		}
+		// Create a range string, e.g. "1-5"
+		pageRange := fmt.Sprintf("%d-%d", i+1, end)
+		outputFilePath := filepath.Join(tempDir, fmt.Sprintf("output-%d-%d.pdf", i+1, end))
+
+		// Extract the page range using TrimFile instead of ExtractPagesFile
+		if err := api.TrimFile(inputFilePath, outputFilePath, []string{pageRange}, nil); err != nil {
```

Comment:
❓ Is `nil` intentional here? Should we use `model.NewDefaultConfiguration()` as in [here](https://github.com/pdfcpu/pdfcpu/blob/master/pkg/api/api.go#L85C35-L85C58)?